lib/Console/Traits/Parallelization.php

```diff
@@ -64,6 +68,22 @@ protected static function configureParallelization(Command $command): void
         ;
     }
 
+    protected function getParallelExecutableFactory(
+        callable $fetchItems,
+        callable $runSingleCommand,
+        callable $getItemName,
+        string $commandName,
+        InputDefinition $commandDefinition,
+        ErrorHandler $errorHandler
+    ): ParallelExecutorFactory {
+        return ParallelExecutorFactory::create(...func_get_args())
+            ->withSegmentSize($this->getSegmentSize())
+            ->withScriptPath($this->getConsolePath())
```

Comment:
is this (and the method `getConsolePath()`) really necessary? I think it should be that one picked by default and then if the user wants to use another one he can configure it like you are doing it, i.e. you can phase out that `getConsolePath()` method.

To double check with the last release there should be a more friendly upgrade path, at least it keeps most of the old parallelization trait API but points out what has been deprecated

Reply:
@dvesh3  Is there a specific reason, why the segment size is the batch size?
```php
    protected function getSegmentSize(): int
    {
        return (int)$this->input->getOption('batch-size');
    }
```
This is kind of confusing since the segment size would actually be an own option. Is it just because there is a default value of 50 for the batch size?

@theofidry would it be better if we add another option for the script path but by default use ```/bin/console```

Reply:
>Is there a specific reason, why the segment size is the batch size?

By default (coming from 1.x), the bach size was at `50` and the segment size was `getBatchSize()` (so `50` as well by default). I am not entirely sure of the reason, I think the intent was:

- assume the app is leaky (it was definitely the case internally when we introduced this trait before this was even a lib), so we could not really afford to have a segment size greater than the batch size at least in the common case (at that time)
- if you bump batch size the segment size should remain bigger than the batch size

So although not perfect, I reckon with that context this was "good enough" at the time.

Fast forward in 2.x:

- I kept those defaults although they are now in `ParallelExecutorFactory`
- Via the `ParallelExecutorFactory`, it can be configured directly at the class level, which is automatically done if you are overriding `getBatchSize()` or `getSegmentSize()` in the `Parallelization` trait (but you should get a deprecation notice warning you to configure the `ParallelExecutorFactory` directly instead)
- Via the input options

**Maybe the default could value of the segment size could be revisited however.**

>This is kind of confusing since the segment size would actually be an own option.

I believe the excerpt you gave is from your code not the parallelization library unless I remember wrong. So I agree this looks confusion, and no longer makes sense in 2.x at the very least. Could it be it was there in 1.x because at that time there was no such options?

>would it be better if we add another option for the script path but by default use /bin/console

I think the native extension point should be good enough (the `{get|configure}ParallelExecutorFactory()`).

Also note that now (in 2.x), the script path is actually detected automatically so I think the % of users that actually want to override it now is close to 0% (although still possible – and allowed).